PURL-TYPES.rst

```diff
@@ -255,6 +255,37 @@ rpm
       pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
       pkg:rpm/centerim@4.22.10-1.el6?arch=i686&epoch=1&distro=fedora-25
 
+swift
+----
+``swift`` for Swift packages:
+
+- There is currently no default repository for Swift packages.
+- The ``namespace`` is used for the scope of a Swift package.
+  It is not case sensitive and must be lowercased.
+
+  - A package scope consists of alphanumeric characters and hyphens.
+    Hyphens may not occur at the beginning or end,
+    nor consecutively within a scope.
+  - The maximum length of a package scope is 39 characters.
+  - A valid package scope matches the following regular expression pattern
+    ``\A[a-zA-Z0-9](?:[a-zA-Z0-9]|-(?=[a-zA-Z0-9])){0,38}\z``
+
+- The ``name`` is the name declared in the Swift package manifest.
+  It is case and normalization insensitive,
+  and must be normalized using Unicode Normalization Form KC (NFKC)
+  and then transformed with locale-independent case folding.
+
+  - The maximum length of a package name is 128 characters.
+  - A valid package name matches the following regular expression pattern
+    ``\A\p{XID_Start}\p{XID_Continue}{0,127}\z``
```

Comment:
IMO if we want to specify a regex pattern for a valid package name, we need to account for the lowest common denominator. XID_Start and XID_Continue are not recognized as valid patterns in the first two regular expression libraries I used. Many regex parsers only support a small subset of capabilities and others will lean towards one dialect of regex syntax. For simplicity, I would recommend using english language to specify what is valid and what is not.

Reply:
The regular expression here uses the `\p{UNICODE PROPERTY NAME}` metacharacter described by [ICU pattern syntax](https://unicode-org.github.io/icu/userguide/strings/regexp.html). Unfortunately, it's not feasible to express this using an alternative syntax (see [UAX #31](https://unicode.org/reports/tr31/)).

What do you think about something like this?

> - A valid package name consists of a character with the Unicode property `XID_Start` followed by up to 127 characters with the Unicode property `XID_Continue`, or expressed by ICU regular expression pattern:
     `\A\p{XID_Start}\p{XID_Continue}{0,127}\z`